doc/compiler/optimizer/InliningAndInliningGuards.md

```diff
@@ -0,0 +1,406 @@
+3. Overview of Guards
+4. Different Types Of Guards
+
+# 1. The Benefits of Inlining
+
+Inliner, local CSE (Common Subexpression Elimination), GRA (Global Register Allocation), and Global Value Propagation
+are some of the most impactful optimizations from the perspective of improving performance throughput.
+These highly effective optimizations are important for scorching and hot compilations as well as for a flat
+profile with warm compilations.
+
+## Basic Concepts
+- **Call site** ([TR_CallSite](https://github.com/eclipse/omr/blob/88427d6420f5fa1bf00f897d1f589390bf2cafae/compiler/optimizer/CallInfo.hpp#L268))
+is the code that makes the call. It contains the call node (caller) that makes the call and the call target
+(callee) that is inlined into the call site.
+- **Call target** ([TR_CallTarget](https://github.com/eclipse/omr/blob/88427d6420f5fa1bf00f897d1f589390bf2cafae/compiler/optimizer/CallInfo.hpp#L130))
+represents the callee or the specific method that is inlined into the call site.
+
+## Eliminate Calling Overhead
+
+The benefits of inlining are wide ranging. An application with a flat profile generally has a lot of compiled
+methods. The calling overhead in a flat profile could add up across all methods. The most direct benefit of
+inlining is to get rid of call overhead such as call instruction, return instruction, prologue, epilogue, etc.
+**The key question when it comes to inlining is how much is going to be executed versus how much code there is
+in the callee**. For example, if a method calls another method that executes a thousand instructions, the call
+overhead is probably not significant. However, if the method that is being called executes only 10 instructions,
+or even just one or two instructions, or if a method returns a constant or returns a simple field read, the cost
+of call instruction, return instruction, prologue, and epilogue adds up and can be substantial, compared to the
+amount of the code that is actually executed.
+
+## Allow Other Optimizations To Be Applied To The Inlined Code
+
+Getting rid of call related overhead is only one aspect of the benefits of inlining. Inlining also exposes larger
+blocks of code to the optimizer which provides more opportunities for other optimizations to improve the program.
+Driven by the expectation that the benefits of other optimizations such as constant folding, branch simplification,
+etc. that could occur, inlining enables JIT to do other powerful optimizations downstream of inlining.
+
+For example, if an inlined method passes a constant from the caller to the callee and the constant is used on
+the callee side, the code can be simplified dramatically by folding away the branches or changing loads into
+constants. Such as when passing a constant `String` from a caller to a callee and `hashCode()` or `equal()` is
+called on the constant `String` in the callee, these operations can be completely folded away during the compile
```

Comment:
Typos here : 
"the compile time" -> "compile time" 
and
"in runtime" -> "at run time"